In [1]:
import requests

In [2]:
response = requests.get("https://news.naver.com/section/105")

In [3]:
data = response.text.split('data-template-id="SECTION_HEADLINE">')[1]
data

'\n\t<div class="sa_head">\n\t\t<span class="sa_head_inner">\n\t\t\t<a href="#" class="sa_head_link _TOGGLE" data-classValue="is_hidden" data-target="_SECTION_HEADLINE_INFO_d4qsm" data-clk="guide">헤드라인 뉴스 <i class="sa_head_icon">안내</i></a>\n\t\t</span>\n\t\t<div id="_SECTION_HEADLINE_INFO_d4qsm" class="sa_head_layer is_hidden">\n\t\t\t<p class="sa_head_layer_p">각 헤드라인의 기사와 배열 순서는 개인화를 반영해 추천되며, 기사묶음의 대표 기사는 구독 언론사 중심으로 제공됩니다. 기사 수량이 표기된 기사 우측 하단의 파란색 아이콘을 클릭하면 기사묶음을 확인할 수 있고 기사묶음과 기사묶음 타이틀도 기사 내용을 기반으로 자동 추출됩니다.</p>\n\t\t\t<a href="https://media.naver.com/algorithm" class="sa_head_layer_go">알고리즘 자세히 보기</a>\n\t\t\t<div class="sa_head_layer_close">\n\t\t\t\t<button type="button" class="_TOGGLE" data-classValue="is_hidden" data-target="_SECTION_HEADLINE_INFO_d4qsm">닫기</button>\n\t\t\t</div>\n\t\t</div>\n\t</div>\n\t<ul id="_SECTION_HEADLINE_LIST_d4qsm" class="sa_list">\n\t\t<li class="sa_item _SECTION_HEADLINE">\n\t\t\t<div class="sa_item_inner">\n\t\t\t\t<div class="sa_item_flex _LAZY_LO

In [4]:
data = data.split('sa_list">')[1]
data = data.split(r'</ul>')[0]
data

'\n\t\t<li class="sa_item _SECTION_HEADLINE">\n\t\t\t<div class="sa_item_inner">\n\t\t\t\t<div class="sa_item_flex _LAZY_LOADING_WRAP">\n\t\t\t\t\t<div class="sa_thumb _LAZY_LOADING_ERROR_HIDE">\n\t\t\t\t\t\t<div class="sa_thumb_inner">\n\t\t\t\t\t\t\t<a href="https://n.news.naver.com/mnews/article/015/0005140284" class="sa_thumb_link _NLOG_IMPRESSION" data-clk="itn.clart" data-imp-gdid="88000107_000000000000000005140284" data-imp-url="https://n.news.naver.com/mnews/article/015/0005140284" data-imp-index="1">\n\t\t\t\t\t\t\t\t<img data-src="https://mimgnews.pstatic.net/image/origin/015/2025/06/04/5140284.jpg?type&#x3D;nf220_150" class="_LAZY_LOADING _LAZY_LOADING_INIT_HIDE" width="110" height="75" alt="최기영 스노우플레이크 지사장 &quot;AI 전환 위해 개발자에 대한 의존도 줄여야&quot;" style="display: none;">\n\t\t\t\t\t\t\t</a>\n\t\t\t\t\t\t</div>\n\t\t\t\t\t</div>\n\t\t\t\t\t<div class="sa_text">\n\t\t\t\t\t\t<a href="https://n.news.naver.com/mnews/article/015/0005140284" class="sa_text_title _NLOG_IMPRESSION" dat

<img class="_LAZY_LOADING _LAZY_LOADING_INIT_HIDE" width="110" height="75" alt="카이스트 사족보행 로봇 '라이보', 고속 질주에 벽까지 달린다" style="" src="https://mimgnews.pstatic.net/image/origin/421/2025/06/03/8291585.jpg?type=nf220_150">

In [8]:
import re
p = re.compile(r"<img.+?src=['\"][^'\"]+['\"][^>]*>")
m = p.search(data)
m

<re.Match object; span=(489, 746), match='<img data-src="https://mimgnews.pstatic.net/image>

In [9]:
images = p.findall(data)
images

['<img data-src="https://mimgnews.pstatic.net/image/origin/015/2025/06/04/5140284.jpg?type&#x3D;nf220_150" class="_LAZY_LOADING _LAZY_LOADING_INIT_HIDE" width="110" height="75" alt="최기영 스노우플레이크 지사장 &quot;AI 전환 위해 개발자에 대한 의존도 줄여야&quot;" style="display: none;">',
 '<img data-src="https://mimgnews.pstatic.net/image/origin/001/2025/06/04/15430668.jpg?type&#x3D;nf220_150" class="_LAZY_LOADING _LAZY_LOADING_INIT_HIDE" width="110" height="75" alt="삼성, &#x27;울트라급&#x27; 폴더블 신작 &#x27;갤럭시 Z 폴드7&#x27; 내달 출시 예고" style="display: none;">',
 '<img data-src="https://mimgnews.pstatic.net/image/origin/003/2025/06/04/13284628.jpg?type&#x3D;nf220_150" class="_LAZY_LOADING _LAZY_LOADING_INIT_HIDE" width="110" height="75" alt="카카오, 중소사업자 대상 개인정보 보호 온라인 교육" style="display: none;">',
 '<img data-src="https://mimgnews.pstatic.net/image/origin/016/2025/06/04/2480401.jpg?type&#x3D;nf220_150" class="_LAZY_LOADING _LAZY_LOADING_INIT_HIDE" width="110" height="75" alt="사람 뇌 닮은 ‘비전 센서’…자율주행차 더 똑똑해진다" style="display: n

In [10]:
iter = p.finditer(data)
iter

In [15]:
next(iter)

StopIteration: 